TumbleBitSetup/PermutationTest/PermutationTestSetup.cs

```diff
@@ -0,0 +1,56 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace TumbleBitSetup
+{
+    public class PermutationTestSetup
+    {
+        public PermutationTestSetup()
+        {
+
+        }
+
+        public PermutationTestSetup(byte[] publicString, int alpha, int keySize, int k = 128)
```

Comment:
Just want to confirm with you guys @osagga @nicolasdorier, based on an earlier github discussion between you.
keySize should NOT be optional in the setup for both PoupardStern and PermutationTest.  (It does not look optional to me, but maybe I am missing something?) keySize must be specified with the setup.  And, as you have done, it must be passed from the setup to verification algorithm, NOT derived from the public key that is given to the verifier.

Why? Because an adversarial Tumbler could give the user a (say) 512 bit public key when the verifier "expects" a 2048 bit one, which would be horrible.  We need to stop this attack. That's why we should not derive the keylen from the pub key given from the verifier, but rather from the setup. 

Just confirm again to me that this is what is happening in this code. (I think it is.)